HealthVault/Classes/Types/MHVThingQuery.m

```diff
@@ -189,20 +202,20 @@ - (void)serialize:(XWriter *)writer
     //
     // Query xml schema says - ids are a choice element
     //
-    if (![MHVCollection isNilOrEmpty:self.thingIDs])
+    if (![NSArray isNilOrEmpty:self.thingIDs])
     {
-        [writer writeElementArray:c_element_id elements:self.thingIDs.toArray];
+        [writer writeElementArray:c_element_id elements:self.thingIDs];
     }
-    else if (![MHVCollection isNilOrEmpty:self.keys])
+    else if (![NSArray isNilOrEmpty:self.keys])
     {
-        [writer writeElementArray:c_element_key elements:self.keys.toArray];
+        [writer writeElementArray:c_element_key thingName:c_element_thing_id elements:self.keys];
     }
```

Comment:
@namalu There's a bug here, it's not serializing the thing keys right.  You can reproduce by turning off caching, and then scrolling down on a page so it reloads the next batch.  Flying out tomorrow morning, and ran out of time to get it fixed. :(  Not serializing the right thing, so the query fails on HV.